File: src/main/java/org/homework20241215/house/House.java
package org.homework20241215.house;

public class House {

    public static HouseBuilder builder() {
        return new HouseBuilder();
    }

    private final String address;
    private final Integer floors;
    private final String floorMaterial;
    private final String wallMaterial;

    public House(String adress, Integer floors, String floorMaterial, String wallMaterial) {
        this.address = adress;
        this.floors = floors;
        this.floorMaterial = floorMaterial;
        this.wallMaterial = wallMaterial;
    }

    public void printHouseDetail() {
        System.out.println(address + "の家");
        if(floors != null) {
            System.out.println("階数:" + floors);
        }
        if(floorMaterial != null) {
            System.out.println("床材:" + floorMaterial);
        }
        if(wallMaterial != null) {
            System.out.println("壁材:" + wallMaterial);
        }
    }
}

File: src/main/java/org/homework20241222/house/House.kt
package org.homework20241222.house;

class House(private val address: String, private val floors: Int, private val floorMaterial: String, private val wallMaterial: String) {

    //companion object {}で囲うとbuilderが使用可能になる,Javaでいう所のstaticのようなもの?
    companion object {
        fun builder(): HouseBuilder {
            return HouseBuilder();
        }
    }

    fun printHouseDetail() {
        println(address + "の家")
        // floors != nullの時の処理
        floors?.let {
            println("階数:" + floors)
        }
        floorMaterial?.let {
            println("床材:" + floorMaterial)
        }
        wallMaterial?.let {
            println("壁材:" + wallMaterial)
        }
    }
}